packages/device_info/android/src/main/java/io/flutter/plugins/deviceinfo/DeviceInfoPlugin.java

```diff
@@ -0,0 +1,77 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+package io.flutter.plugins.deviceinfo;
+
+import android.os.Build;
+import android.os.Build.VERSION;
+import android.os.Build.VERSION_CODES;
+import io.flutter.plugin.common.MethodCall;
+import io.flutter.plugin.common.MethodChannel;
+import io.flutter.plugin.common.MethodChannel.MethodCallHandler;
+import io.flutter.plugin.common.MethodChannel.Result;
+import io.flutter.plugin.common.PluginRegistry.Registrar;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+/** DeviceInfoPlugin */
+public class DeviceInfoPlugin implements MethodCallHandler {
+
+  /** Substitute for missing values. */
+  private static final String[] EMPTY_STRING_LIST = new String[] {};
+
+  /** Plugin registration. */
+  public static void registerWith(Registrar registrar) {
+    final MethodChannel channel =
+        new MethodChannel(registrar.messenger(), "plugins.flutter.io/device_info");
+    channel.setMethodCallHandler(new DeviceInfoPlugin());
+  }
+
+  @Override
+  public void onMethodCall(MethodCall call, Result result) {
```

Comment:
[There is](https://stackoverflow.com/questions/15804365/is-there-any-way-to-get-gpu-information) but not sure how reliable it is. Also, with the exact device model it should be easy to look it up.